File: Pythonleetcode/Leetcode198.py
class Solution(object):

    def rob(self, nums):


        if nums == []:
            return 0

        if len(nums) == 1:
            return max(nums)

        dp = [0] * len(nums)

        dp[0] = nums[0]

        dp[1] = max(nums[1], nums[0])

        for i in range(2, len(nums)):
            dp[i] = max(dp[i - 1], dp[i - 2] + nums[i])

        return dp[len(nums) - 1]


print(Solution().rob([2,1,1,2]))

File: CppPractice/CppLeetcode/Leetcode198.cpp
class Solution {
public:
    int rob(vector<int>& nums) 
    {
        vector<int> dp = vector<int>(nums.size()+2,0);
        for(int i=2;i<dp.size();i++)
        {
            dp[i]=max(dp[i-1],dp[i-2]+nums[i-2]);
        }
        return dp[dp.size()-1];
    }
};